dev/com.ibm.ws.componenttest.2.0/src/componenttest/app/AssertionErrorSerializer.java

```diff
@@ -0,0 +1,216 @@
+    //JSON property names
+    private static final String CLASS_NAME_KEY = "className";
+    private static final String METHOD_NAME_KEY = "methodName";
+    private static final String EXCEPTION_TYPE_KEY = "exceptionType";
+    private static final String MESSAGE_KEY = "message";
+    private static final String STACK_KEY = "stack";
+    private static final String FILE_NAME_KEY = "fileName";
+    private static final String LINE_NUMBER_KEY = "lineNumber";
+    //The FATServlet.doGet method name
+    private static final String DO_GET = "doGet";
+
+    /**
+     * Simplify an AssertionError that has been thrown from a test in a subclass of FATServlet.
+     * This will change the error message to include the FAT classname and test method name.
+     * It will also shorten the stacktrace to stop at FATServlet.doGet because everything after that
+     * will always be the same and is not of any interest.
+     *
+     * @param fatClass      The FAT class which originally threw the AssertionError
+     * @param fatMethodName The FAT method which originally threw the AssertionError
+     * @param e             The original AssertionError
+     * @return A simplified AssertionError
+     */
+    public static <T extends FATServlet> AssertionError simplify(Class<T> fatClass, String fatMethodName, AssertionError e) {
+        AssertionError assertionError = new AssertionError(fatClass.getSimpleName() + "." + fatMethodName + ": " + e.getMessage());
+        StackTraceElement[] originalStack = e.getStackTrace();
+        ArrayList<StackTraceElement> shortenedStack = new ArrayList<>();
+
+        for (StackTraceElement element : originalStack) {
+            String declaringClass = element.getClassName();
+            String methodName = element.getMethodName();
+            String fileName = element.getFileName();
+            int lineNumber = element.getLineNumber();
+            StackTraceElement newElement = new StackTraceElement(declaringClass, methodName, fileName, lineNumber);
+
+            shortenedStack.add(newElement);
+
+            //the stack beyond doGet is always the same so don't output any more
+            if ((FATServlet.class.getSimpleName() + ".java").equals(fileName) && DO_GET.equals(methodName)) {
+                break;
+            }
+        }
+
+        assertionError.setStackTrace(shortenedStack.toArray(new StackTraceElement[shortenedStack.size()]));
+        return assertionError;
+    }
+
+    /**
+     * Serialize out an AssertionError as a JSON String
+     *
+     * @param e The AssertionError
+     * @return A JSON String
+     */
+    public static String serialize(AssertionError e) {
+        JsonObject json = serializeAssertionError(e);
+        StringWriter writer = new StringWriter();
+        JsonWriter jsonWriter = Json.createWriter(writer);
+        jsonWriter.writeObject(json);
+        jsonWriter.close();
+        return writer.toString();
+    }
```

Comment:
It's probably better to pass in the `writer` to use, rather than creating a `StringWriter`, returning a `String` and then writing that `String` to a different writer.

Reply:
ok, I had it that way to start with but wanted to make the serialize and deserialize methods symetrical. Will look again to see if there is a better way.

Reply:
I overloaded the methods so both forms are available. FATServlet now passes in a Writer.